components/IndoorDirections/TypesOfDirections/DifferentBuildingDirections.js

```diff
@@ -1,25 +1,31 @@
 import React from 'react';
 import { Line, G } from 'react-native-svg';
 import { ClassGraph } from '../../../constants/ClassGraph';
+import { Class9Graph } from '../../../constants/ClassGraph9';
 import { dijkstra, getFloorNumber, ConvertToHall8Floor } from '../Dijkstra/DijkstraAlgorithm';
+import { Hall9Coordinates } from '../../../constants/Hall9Coordinates';
 
 export function DifferentBuildingDirections(props) {
-    const rooms = props.rooms;
-    const graph = ClassGraph();
+    var floorFrom;
+    var floorTo;
+    var rooms = props.rooms;
+    var graph = ClassGraph(); 
     var goTo = props.from;
     var tempGoTo = props.from;
 
     if (props.from.includes(" ")) {
-        tempGoTo = goTo;
-        goTo = ConvertToHall8Floor(props.to);
+        floorTo = getFloorNumber(props.to);
+        tempGoTo = props.to;
+        goTo = floorTo == 9 ? props.to : ConvertToHall8Floor(props.to);
     }
 
     if (props.to.includes(" ")) {
-        tempGoTo = goTo;
-        goTo = ConvertToHall8Floor(props.from);
+        floorFrom = getFloorNumber(props.from);
+        tempGoTo = props.from;
+        goTo = floorFrom == 9 ? props.from : ConvertToHall8Floor(props.from);
```

Comment:
I see what you have done here with ternary statements, but in a case where we would have a lot of different floors, I think this solution would demand a lot condition nesting, correct me, if i'm wrong.  And also, do we need a convertToHall5Floor, since we also have that specific floor for the traced route?

Reply:
Hall floor 5 is the 8th floor of H. For now, we only have 2 floor plans. If the requirements asks us for more, I will do a refactor of merging everything into one file